### PR TITLE
feat: Use exiting connection

### PR DESCRIPTION
**Pull request by reinaldooli**
Allows to pass an existing connection so the adapter can use that connection instead of creating a new one.

Update casbin and typeorm.

### CODE REVIEW

**Comment by hsluoyz** [comment]
@nodece

**Comment by nodece** [comment]
@reinaldooli Thanks for your contribution, here are some things to do:

- Add the `index.ts` file for export `CasbinMongoRule`,  `CasbinRule` and `TypeORMAdapter`, `TypeORMAdapter` should be `export default`
- The new version should be `1.2.1` or `1.3.0`

**Comment by reinaldooli** [comment]
@nodece thank you for the tips.

I have made the corrections and fixed the tests to use MySQL connection again.

If you have any other advices let me know.

👑

**Comment by hsluoyz** [comment]
@nodece

**Comment by hsluoyz** [comment]
Fix CI.

**Comment by reinaldooli** [comment]
chalk@^4 is required by new version of Typeorm.
What you think I can do? Downgrade Typeorm?

This CI step for Node 8 is really required?

**Comment by hsluoyz** [comment]
@nodece what do you think about the Node 8 compatibility?

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "typeorm-adapter",
-  "version": "1.2.0",
+  "version": "1.3.0",
   "description": "TypeORM adapter for Casbin",
   "main": "lib/adapter.js",
```
Comment (nodece) [inline package.json:5]: should be `lib/index.js`.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
   "typings": "lib/adapter.d.ts",
```
Comment (nodece) [inline package.json:6]: should be `lib/index.d.js`.

File: src/index.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export * from './adapter';
+export * from './casbinMongoRule';
+export * from './casbinRule';
```
Comment (nodece) [inline src/index.ts:3]: There must be `export default` here for compatible with previous version.

It looks like this,
```ts
export { TypeORMAdapterOptions } from './adapter';
export * from './casbinMongoRule';
export * from './casbinRule';

import TypeORMAdapter from './adapter';

export default TypeORMAdapter;
```

File: test/adapter.test.ts [hunk 4]
```diff
@@ -12,129 +12,149 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import {Enforcer, Util} from 'casbin';
+import { Enforcer, Util } from 'casbin';
 import TypeORMAdapter from '../src/adapter';
```
Comment (nodece) [inline test/adapter.test.ts:16]: import TypeORMAdapter from '../src/index';

File: test/existent-connection-adapter.test.ts [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Enforcer, Util } from 'casbin';
+import { createConnection } from 'typeorm';
+import TypeORMAdapter from '../src/adapter';
+import { CasbinRule } from '../src/casbinRule';
```
Comment (nodece) [inline test/existent-connection-adapter.test.ts:18]: import TypeORMAdapter, { CasbinRule } from '../src/index';

**Comment by nodece** [comment]
> chalk@^4 is required by new version of Typeorm.
> What you think I can do? Downgrade Typeorm?
> 
> This CI step for Node 8 is really required?

@reinaldooli Could you try the `yarn install --ignore-engines` on TravisCI? If it still fails, you need to delete node@8 from TravisCI.

**Review by nodece — APPROVED** [review]

**Pull request merged**